velox/functions/prestosql/Zip.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <boost/algorithm/string/join.hpp>
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/lib/LambdaFunctionUtil.h"
+
+namespace facebook::velox::functions {
+
+namespace {
+void validateType(const std::vector<VectorPtr>& inputArgs) {
+  VELOX_USER_CHECK_GT(
+      inputArgs.size(), 1, "zip requires at least two parameters");
+
+  for (auto& arg : inputArgs) {
+    VELOX_USER_CHECK_EQ(
+        arg->type()->kind(),
+        TypeKind::ARRAY,
+        "zip requires arguments of type ARRAY");
+  }
+}
+
+class ZipArrayFunction : public exec::VectorFunction {
+  const static auto MIN_ARITY = 2;
+  const static auto MAX_ARITY = 7;
+
+ public:
+  /// This class implements the zip function.
+  ///
+  /// DEFINITION:
+  /// zip is defined as follows :
+  /// zip(ARRAY[T], ARRAY[U]) -> ARRAY(ROW[T,U])
+  /// where we create a ROW[Ti, Ui] for every ith element in ARRAY[T], ARRAY[U].
+  /// The smaller array is padded with Nulls.
+  ///
+  /// IMPLEMENTATION:
+  ///  1. The general idea is to create a new dictionary vector for each input
+  ///  array vector and enumerating their indexes to create a 1:1 mapping.
+  ///  2. To do this for each row we will determine which is the largest Array
+  ///  and subsequently pad the smaller arrays with Nulls to make them all
```

Comment:
- typo: Nulls -> nulls
- "to make them all the same size" is the definition of padding, hence, redundant; consider removing.